Automattic/woocommerce-payments
# Visually hide UPE payment methods if manual capture is enabled

Fixes #3567 

#### Changes proposed in this Pull Request

- We show a confirmation modal when the merchant tries to enable the manual capture while UPE is enabled.
- If manual capture is enabled while UPE is enabled too, then the UPE methods will look like they are disabled on the payment settings screen.
- We add an E2E test file to cover a few scenarios about manual capture in the context of UPE methods

The reason is that UPE methods don't support manual capture so they won't be shown at checkout, and this is currently not clear to the merchants.

This is what it looks like after the changes, when manual capture is enabled:
![image](https://user-images.githubusercontent.com/28830738/167872495-cc191c31-4fcc-47c7-934d-ddf5f4049d53.png)

#### Testing instructions

- On your site, activate UPE and visit Payment > Settings page
- You should see that UPE methods can be enabled/disabled and appear just like the `Credit card / Debit card` method
- Check the "Issue an authorization on checkout, and capture later" checkbox
- A confirmation modal is shown with the relevant text
- If you cancel, then nothing has changed for the appearance of the payment methods
- If you try again to check the checkbox and confirm your choice, you should now see that the appearance of UPE methods has changed
  - instead of a checkbox to enable/disable, there's a warning icon which shows a tooltip when hovering over it
  - there's an overlay that indicates that it is not "available"
- If you uncheck the checkbox, you should not see a confirmation modal and the appearance should be back to normal for all payment methods

If UPE is disabled, enabling/disabling the manual capture should be immediate, without a confirmation modal.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-4.3.0
